### PR TITLE
Virt-who general configuration

### PR DESCRIPTION
**Pull request by cnsnyder**
This PR implements the design for general configuration located here:
http://etherpad.corp.redhat.com/virt-who-general-conf

### CODE REVIEW

**Comment by cnsnyder** [comment]
After review, I will squash the commits down to smaller more manageable ones.

File: config.py [hunk 1]
```diff
@@ -50,7 +54,127 @@ def strip_quote(s):
...
+        # a reference to the dictionary to appear
+        self.__dict__['_options'] = options
+
+    def __getattr__(self, name):
+        if name.startswith('_'):
+            super(GeneralConfig, self).__getattr__(name)
+
+        value = self._options.get(name, None)
+        if value is None:
+            if name in self.DEFAULTS:
```
Comment (rnovacek) [inline config.py]: Use `return self.DEFAULTS.get(name, None)`.

File: tests/test_config.py [hunk 2]
```diff
@@ -432,3 +434,97 @@ def testMissingOwnerOption(self):
...
+            'test': '1',
+            'override_me':'wrong_value'
+        }
+        arg_dict = {
+            'arg_dict':'some_value',
+            'override_me':'expected_value'
+        }
+        general_config = GeneralConfig(defaults=defaults, **arg_dict)
+
+        self.assertEqual(getattr(general_config, 'test'), '1')
```
Comment (rnovacek) [inline tests/test_config.py]: Why `getattr()` instead of `general_config.test`?

**Comment by rnovacek** [comment]
Great job! I'll merge it right now. Those nitpicks I found are nothing serious, they can be fixed later on.

**Comment by rnovacek** [comment]
Ah, you wanted to squash the commits... never mind...

**Pull request merged**